### PR TITLE
include qmake test into cmake build

### PR DESCRIPTION
**Pull request by gmabey**
This suggestion tracks https://stackoverflow.com/questions/39704341/both-cmake-and-qmake-usage as a means of incorporating a qmake build subproject within a larger cmake build.  In this way, the qmake support can be tested automatically.

### CODE REVIEW

File: tests/test_qmakebuild/main.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Alexey Edelev <semlanik@gmail.com>
```
Comment (semlanik) [inline tests/test_qmakebuild/main.cpp:4]: nit: Recent year and your name here, please :)

File: tests/test_qmakebuild/main.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include <QCoreApplication>
+
+#include <QtProtobufTypes>
+
```
Comment (semlanik) [inline tests/test_qmakebuild/main.cpp:29]: It makes sense to use generated classes there.

File: tests/test_qmakebuild/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#set(TARGET wellknowntypes_test)
+project(qmakebuild_test)
+
```
Comment (semlanik) [inline tests/test_qmakebuild/CMakeLists.txt:3]: Missed
`find_package(Qt5 COMPONENTS Core REQUIRED)`

File: tests/test_qmakebuild/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+get_target_property (QT_QMAKE_EXECUTABLE Qt5::qmake IMPORTED_LOCATION)
```
Comment (semlanik) [inline tests/test_qmakebuild/CMakeLists.txt:4]: CMake accepts using the target as the first argument of "COMMAND" without expanding its path. If it's the second or subsequent argument you may use $<TARGET_FILE:Qt5::qmake> genex directly without getting the imported location.

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -12,6 +12,7 @@ if(TARGET ${QT_VERSIONED_PREFIX}::QuickTest)
 endif()
 add_subdirectory("test_protobuf_multifile")
 add_subdirectory("test_extra_namespace")
+add_subdirectory("test_qmakebuild")
```
Comment (semlanik) [inline tests/CMakeLists.txt:15]: I'm not quite sure if the test will work without QtProtobuf installer in the system.
I'm referring the https://github.com/semlanik/qtprotobuf/pull/225/files#diff-2167d9f28c33778cbc1f40d89aef2642ac5a795779df44ed6d07852fc5049758R2 . AFAIK this doesn't work if .pri files are not inside mkspec directory.

So my suggestion is to put the test into
```cmake
if(QT_PROTOBUF_STANDALONE_TESTS)
    add_subdirectory("test_qmakebuild")
endif()
```

File: tests/test_qmakebuild/test1.proto [hunk 6]
```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
```
Comment (semlanik) [inline tests/test_qmakebuild/test1.proto:1]: I will create a 'sanity' subset of proto messages to do not copy it each time later :)

**Review by semlanik — COMMENTED** [review]
I think you are tired of me constantly breaking .pri and qmake builds

File: tests/test_qmakebuild/CMakeLists.txt (hunk 4 above)
Comment (gmabey) [reply tests/test_qmakebuild/CMakeLists.txt:4]: I haven't been able to make this work yet.  Specifically, I tried
```
COMMAND ${TARGET_FILE:Qt5::qmake} -o QMakefile
```
and got 
```
Invalid character (':') in a variable name: 'TARGET_FILE'
```

Comment (semlanik) [reply tests/test_qmakebuild/CMakeLists.txt:4]: genexes use `<` and `>` instead of `{` and `}`

File: tests/CMakeLists.txt (hunk 5 above)
Comment (gmabey) [reply tests/CMakeLists.txt:15]: I'll propose another option in my next PR.

**Comment by gmabey** [comment]
hmm -- the builds sure don't like this setup ...

**Comment by semlanik** [comment]
I will check what's wrong locally and back with a suggestion.

**Comment by semlanik** [comment]
@gmabey I checked the approach quickly and it doesn't seem to work from the qt build tree. The reason for that is hardcoded paths to the QtProtobuf libraries that supposed to be resolved only after installing the step. So perhaps this should be postponed till I merge standalone testing.

**Comment by gmabey** [comment]
That's fine to postpone it -- I'll try to look at it again here soon, and see if there's something that can be done.

**Pull request closed without merging**